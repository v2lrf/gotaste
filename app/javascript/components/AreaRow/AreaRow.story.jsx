import React from 'react'
import { storiesOf } from '@storybook/react'

import { AreaRow } from '.'

const fakeArea = {
  name: 'Indre By',
  businesses: [
    {
      id: '1',
      name: 'Test',
      streetName: 'Baggesensgade',
      streetNumber: '13',
      postalCode: '2200',
      city: 'København N',
      slug: 'test'
    },
    {
      id: '2',
      name: 'Foo',
      streetName: 'Bar Street',
      streetNumber: '123',
      postalCode: '2300',
      city: 'København S',
      slug: 'foo'
    }
  ]
}

storiesOf('AreaRow', module).add('Default', () => (
  <div className="p-16">
    <AreaRow {...fakeArea} />
  </div>
))
