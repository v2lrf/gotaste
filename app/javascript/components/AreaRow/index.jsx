import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import { BusinessCard } from '../BusinessCard'
import { Row, Col } from '../Grid'
import Routes from '../../services/Routes'

export function AreaRow({ name, businesses }) {
  return (
    <Fragment>
      <h2 className="text-center mb-6 text-red-dark">{name}</h2>
      <Row>
        {businesses.map(business => (
          <Col xs="full" sm="1/2" lg="1/3" key={business.id}>
            <BusinessCard
              href={Routes.business_path(business.slug)}
              {...business}
            />
          </Col>
        ))}
      </Row>
    </Fragment>
  )
}

AreaRow.propTypes = {
  name: PropTypes.string.isRequired,
  businesses: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      streetName: PropTypes.string.isRequired,
      streetNumber: PropTypes.string.isRequired,
      postalCode: PropTypes.string.isRequired,
      city: PropTypes.string.isRequired,
      slug: PropTypes.string.isRequired
    })
  )
}

AreaRow.defaultProps = {
  businesses: []
}
export default AreaRow
