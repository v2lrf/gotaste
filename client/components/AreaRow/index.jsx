import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import BusinessCard from '../BusinessCard'
import { Row, Col } from '../Grid'

function AreaRow({ name, businesses }) {
  return (
    <Fragment>
      <h3 className="text-center mb-4 text-red-darker">{name}</h3>
      <Row>
        {businesses.map(business => (
          <Col xs="full" sm="1/2" lg="1/3" key={business.id}>
            <BusinessCard {...business} />
          </Col>
        ))}
      </Row>
    </Fragment>
  )
}

AreaRow.propTypes = {
  name: PropTypes.string.isRequired,
  businesses: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      address: PropTypes.shape({
        streetName: PropTypes.string.isRequired,
        streetNumber: PropTypes.string.isRequired,
        postalCode: PropTypes.string.isRequired,
        city: PropTypes.string.isRequired
      }).isRequired,
      slug: PropTypes.string.isRequired
    })
  )
}

AreaRow.defaultProps = {
  businesses: []
}

export default AreaRow
