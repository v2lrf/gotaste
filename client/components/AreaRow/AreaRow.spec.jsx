import React from 'react'
import AreaRow from '.'

describe('AreaRow', () => {
  let component
  let props
  beforeEach(() => {
    props = {
      name: 'Indre By',
      businesses: [
        {
          id: '1',
          name: 'Test',
          address: {
            streetName: 'Baggesensgade',
            streetNumber: '13',
            postalCode: '2200',
            city: 'København N'
          },
          slug: 'test',
          logoId: 'logoId1',
          heroImageId: 'heroImageId1'
        },
        {
          id: '2',
          name: 'Foo',
          address: {
            streetName: 'Bar Street',
            streetNumber: '123',
            postalCode: '2300',
            city: 'København S'
          },
          slug: 'foo',
          logoId: 'logoId2',
          heroImageId: 'heroImageId2'
        }
      ]
    }

    component = render(<AreaRow {...props} />)
  })

  describe('Snapshots', () => {
    it('should match the snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
