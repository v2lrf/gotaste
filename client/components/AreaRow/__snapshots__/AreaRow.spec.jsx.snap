// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AreaRow Snapshots should match the snapshot 1`] = `
Array [
  <h2
    class="text-center mb-6 text-red-dark"
  >
    Indre By
  </h2>,
  <div
    class="px-2"
  >
    <div
      class="flex flex-wrap -mx-2"
    >
      <div
        class="px-2 w-full sm:w-1/2 lg:w-1/3"
      >
        <a
          class="inline-flex flex-col rounded shadow no-underline hover:shadow-md mb-4"
          href="/business/test"
        >
          <div>
            <img
              class="rounded-t"
              height="200"
              src="https://res.cloudinary.com/dkrjpli3y/image/upload/c_fill,h_200,w_600/heroImageId1"
              width="600"
            />
          </div>
          <div
            class="p-4"
          >
            <div
              class="flex items-center"
            >
              <figure
                class="flex flex-no-shrink self-start mr-2 bg-white shadow rounded-full w-12 h-12"
              >
                <img
                  alt=""
                  class="rounded-full"
                  src="https://res.cloudinary.com/dkrjpli3y/image/upload/c_scale,w_200/logoId1"
                  width="200"
                />
              </figure>
              <div
                class="flex-grow"
              >
                <div
                  class="text-black font-bold text-base"
                >
                  Test
                </div>
                <div
                  class="text-grey-darker text-sm"
                >
                  Baggesensgade 13, 2200 København N
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
      <div
        class="px-2 w-full sm:w-1/2 lg:w-1/3"
      >
        <a
          class="inline-flex flex-col rounded shadow no-underline hover:shadow-md mb-4"
          href="/business/foo"
        >
          <div>
            <img
              class="rounded-t"
              height="200"
              src="https://res.cloudinary.com/dkrjpli3y/image/upload/c_fill,h_200,w_600/heroImageId2"
              width="600"
            />
          </div>
          <div
            class="p-4"
          >
            <div
              class="flex items-center"
            >
              <figure
                class="flex flex-no-shrink self-start mr-2 bg-white shadow rounded-full w-12 h-12"
              >
                <img
                  alt=""
                  class="rounded-full"
                  src="https://res.cloudinary.com/dkrjpli3y/image/upload/c_scale,w_200/logoId2"
                  width="200"
                />
              </figure>
              <div
                class="flex-grow"
              >
                <div
                  class="text-black font-bold text-base"
                >
                  Foo
                </div>
                <div
                  class="text-grey-darker text-sm"
                >
                  Bar Street 123, 2300 København S
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>,
]
`;
